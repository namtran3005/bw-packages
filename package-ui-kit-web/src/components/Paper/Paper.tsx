import * as React from 'react';
import {
  Root,
  RootProps as PaperProps,
  RootContent,
  shadows,
} from './Paper.styled';

const Paper: React.FunctionComponent<PaperProps> = ({
  children,
  ...cleanProps
}) => (
  <Root {...cleanProps}>
    <RootContent>{children}</RootContent>
  </Root>
);

export { Paper, PaperProps, shadows };
