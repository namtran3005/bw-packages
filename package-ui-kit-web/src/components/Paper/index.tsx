export { Paper, PaperProps, shadows } from './Paper';
