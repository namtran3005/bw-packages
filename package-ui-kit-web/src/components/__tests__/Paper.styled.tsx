import * as React from 'react';
import { shallow } from 'enzyme';
import { Root, RootComponent } from '../Paper/Paper.styled';

describe('Paper', () => {
  it('should render paper', () => {
    const wrapper = shallow(<Root />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render paper with background', () => {
    const wrapper = shallow(<Root background="primaryLilac" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render paper with background alpha', () => {
    const wrapper = shallow(<Root backgroundAlpha={0.5} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render paper with background and alpha', () => {
    const wrapper = shallow(
      <Root background="primaryLilac" backgroundAlpha={0.5} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render paper with spacing', () => {
    const wrapper = shallow(<Root spacing={24} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render rounded paper', () => {
    const wrapper = shallow(<Root rounded />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render highlighted paper', () => {
    const wrapper = shallow(<Root highlight="primaryLilac" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render paper component', () => {
    const wrapper = shallow(<RootComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
