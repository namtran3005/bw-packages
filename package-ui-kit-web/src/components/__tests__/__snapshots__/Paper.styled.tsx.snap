// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper should render highlighted paper 1`] = `
.c0.c0.c0 {
  background: #FFFFFF;
  border-radius: 0;
  border-top: 4px solid #BEAAFF;
  border-radius: none;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  className="c0"
  highlight="primaryLilac"
/>
`;

exports[`Paper should render paper 1`] = `
.c0.c0.c0 {
  background: #FFFFFF;
  border-radius: 0;
  border-radius: none;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  className="c0"
/>
`;

exports[`Paper should render paper component 1`] = `
<WithStyles(ForwardRef(Paper))
  elevation={1}
/>
`;

exports[`Paper should render paper with background 1`] = `
.c0.c0.c0 {
  background: #BEAAFF;
  border-radius: 0;
  border-radius: none;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  background="primaryLilac"
  className="c0"
/>
`;

exports[`Paper should render paper with background alpha 1`] = `
.c0.c0.c0 {
  background: #FFFFFF;
  border-radius: 0;
  border-radius: none;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  backgroundAlpha={0.5}
  className="c0"
/>
`;

exports[`Paper should render paper with background and alpha 1`] = `
.c0.c0.c0 {
  background: rgba(190,170,255,0.5);
  border-radius: 0;
  border-radius: none;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  background="primaryLilac"
  backgroundAlpha={0.5}
  className="c0"
/>
`;

exports[`Paper should render paper with spacing 1`] = `
.c0.c0.c0 {
  background: #FFFFFF;
  border-radius: 0;
  border-radius: none;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  className="c0"
  spacing={24}
/>
`;

exports[`Paper should render rounded paper 1`] = `
.c0.c0.c0 {
  background: #FFFFFF;
  border-radius: 0;
  border-radius: 8px;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.04000000000000001),0px 2px 2px 0px rgba(0,0,0,0.028000000000000004),0px 3px 1px -2px rgba(0,0,0,0.024);
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  overflow: initial;
  z-index: initial;
}

.c0.c0.c0 > .sc-bdfBwQ {
  box-sizing: border-box;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<Paper
  className="c0"
  rounded={true}
/>
`;
